src/gate_detect/src/GateDetector.cpp

```diff
@@ -0,0 +1,362 @@
+/*
+ * Created By: Cameron Newton
+ * Created On: February 27th, 2018
+ * Description: Filter through image and determine if gate is in image and how
+ * far away it is
+ */
+
+#include "GateDetector.h"
+
+GateDetector::GateDetector(int cannyLow,
+                           int houghLinesThreshold,
+                           int houghLinesMinLength,
+                           int houghLinesMaxLineGap,
+                           int poleMax) {
+    cannyLow_ = cannyLow;
+
+    poleMax_ = poleMax;
+
+    houghLinesThreshold_ = houghLinesThreshold;
+
+    houghLinesMinLength_ = houghLinesMinLength;
+
+    houghLinesMaxLineGap_ = houghLinesMaxLineGap;
+
+    lowVertThresh_ = 20;
+
+    lowHorThresh_ = 60;
+
+    VertInterpolationConstant2_ = HorInterpolationConstant2_ = 81.88;
+
+    VertInterpolationConstant1_ = HorInterpolationConstant1_ = -.92791;
+}
+
+GateDetector::GateDetector() {
+    cannyLow_ = 50;
+
+    poleMax_ = 200;
+
+    houghLinesThreshold_ = 50;
+
+    houghLinesMinLength_ = 50;
+
+    houghLinesMaxLineGap_ = 50;
+
+    lowVertThresh_ = 20;
+
+    lowHorThresh_ = 60;
+
+    VertInterpolationConstant2_ = HorInterpolationConstant2_ = 81.88;
+
+    VertInterpolationConstant1_ = HorInterpolationConstant1_ = -.92791;
```

Comment:
We're definitly going to need some way of setting these magic numbers (and they should probably be in a launch file).